compiler/desugared/from_surface.ml

```diff
@@ -1142,6 +1142,32 @@ let process_scope_use_item
   | Surface.Ast.Rule rule -> process_rule precond scope ctxt prgm rule
   | Surface.Ast.Definition def -> process_def precond scope ctxt prgm def
   | Surface.Ast.Assertion ass -> process_assert precond scope ctxt prgm ass
+  | Surface.Ast.DateRounding (r, _) ->
+    let scope_uid = scope in
+    let scope : Ast.scope = ScopeName.Map.find scope_uid prgm.program_scopes in
+    let r =
+      match r with
+      | Surface.Ast.Increasing -> Ast.Increasing
+      | Surface.Ast.Decreasing -> Ast.Decreasing
+    in
+    let new_scope =
+      match
+        List.find_opt
+          (fun scope_opt ->
+            scope_opt = Ast.DateRounding Ast.Increasing
+            || scope_opt = Ast.DateRounding Ast.Decreasing)
+          scope.scope_options
+      with
+      | Some _ ->
+        Errors.raise_spanned_error (Marked.get_mark item)
+          "A date rounding mode has already been specified"
```

Comment:
I know I am being annoying :grimacing: but it would be very nice for the user to show the two positions of the two date rounding specifications here :pray:

Reply:
You're absolutely right. This is done now.